db.js

```diff
@@ -253,46 +253,40 @@ export class ActivityPubDB extends EventTarget {
     await tx.done()
   }
 
-  async * searchNotes ({ attributedTo, inReplyTo, timeline } = {}, { skip = 0, limit = DEFAULT_LIMIT, sort = -1 } = {}) {
+  async * searchNotes ({ timeline, attributedTo, inReplyTo } = {}, { skip = 0, limit = DEFAULT_LIMIT, sort = -1 } = {}) {
     const tx = this.db.transaction(NOTES_STORE, 'readonly')
-    const indexName = timeline ? 'timeline, published' : inReplyTo ? IN_REPLY_TO_FIELD : (attributedTo ? `${ATTRIBUTED_TO_FIELD}, published` : PUBLISHED_FIELD)
-    console.log('Using index:', indexName)
-    const index = tx.store.index(indexName)
-    const direction = sort > 0 ? 'next' : 'prev'
-
-    if (sort === 0) { // Random sort
-      const totalNotes = await index.count()
-      for (let i = 0; i < limit; i++) {
-        const randomSkip = Math.floor(Math.random() * totalNotes)
-        const cursor = await index.openCursor(null, 'next')
-        if (randomSkip > 0) {
-          await cursor.advance(randomSkip)
-        }
-        if (cursor) {
-          yield cursor.value
-        }
-      }
+    let indexName, keyRange
+
+    if (inReplyTo) {
+      indexName = IN_REPLY_TO_FIELD
+      keyRange = IDBKeyRange.only(inReplyTo)
+    } else if (timeline) {
```

Comment:
Don't use the timeline index at all, just use the published index and filter the timeline when iterating